Content.Server/Automod/AutomodManager.cs

```diff
@@ -0,0 +1,260 @@
+﻿using System.Diagnostics;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Content.Server.Database;
+using Content.Shared.Automod;
+using Content.Shared.Database;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Automod;
+
+public sealed class AutomodManager : IAutomodManager, IPostInjectInit
+{
+    [Dependency] private readonly IPrototypeManager _protoMan = default!;
+    [Dependency] private readonly IEntityManager _entMan = default!;
+    [Dependency] private readonly ILogManager _logMan = default!;
+    [Dependency] private readonly IServerDbManager _db = default!;
+
+    private ISawmill _log = default!;
+
+    // Filters
+    private readonly Dictionary<AutomodTarget, Dictionary<Regex, AutomodFilterDef>> _regexFilters = new();
```

Comment:
You are never actually indexing the interior dictionary, and I don't even think you *can* use Regex objects like that. Use a flat list instead.

Reply:
I'm not sure if this is better, but I turned it into a list. [dc06e4b](https://github.com/space-wizards/space-station-14/pull/28570/commits/dc06e4b46d4fc594290ccfd3db2936b9b32590cc)